Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1948,10 +1949,16 @@ extension BrowserViewController: LegacyTabDelegate {
             let creditCardHelper = CreditCardHelper(tab: tab)
             tab.addContentScript(creditCardHelper, name: CreditCardHelper.name())
 
-            creditCardHelper.foundFieldValues = { fieldValues in
-                guard let tabWebView = tab.webView as? TabWebView else { return }
+            creditCardHelper.foundFieldValues = { [weak self] fieldValues in
+                guard let self = self,
+                      let tabWebView = tab.webView as? TabWebView
+                else { return }
 
-                tabWebView.accessoryView?.reloadViewForCardAccessory()
+                // Delay so that this executes AFTER KeyboardHelperDelegate methods
+                DispatchQueue.main.asyncAfter(deadline: .now() + self.reloadAccessoryForCreditCardTimer) {
+                    tabWebView.accessoryView?.reloadViewFor(.creditCard)
+                    tabWebView.reloadInputViews()
+                }
```

Comment:
Step1: Clean this area so you don't have any accessory view timer

Reply:
Infact you don't need a timer, the `foundFieldValues` should be called at correct time 🤔